metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_config.py

```diff
@@ -0,0 +1,198 @@
+    StatefulIngestionConfigBase,
+)
+from datahub.ingestion.source.usage.usage_common import BaseUsageConfig
+
+
+class DremioConnectionConfig(ConfigModel):
+    hostname: Optional[str] = Field(
+        default=None,
+        description="Hostname or IP Address of the Dremio server",
+    )
+
+    port: int = Field(
+        default=9047,
+        description="Port of the Dremio REST API",
+    )
+
+    username: Optional[str] = Field(
+        default=None,
+        description="Dremio username",
+    )
+
+    authentication_method: Optional[str] = Field(
+        default="PAT",
+        description="Authentication method: 'password' or 'PAT' (Personal Access Token)",
+    )
+
+    password: Optional[str] = Field(
+        default=None,
+        description="Dremio password or Personal Access Token",
+    )
+
+    tls: bool = Field(
+        default=True,
+        description="Whether the Dremio REST API port is encrypted",
+    )
+
+    disable_certificate_verification: Optional[bool] = Field(
+        default=False,
+        description="Disable TLS certificate verification",
+    )
+
+    path_to_certificates: str = Field(
+        default=certifi.where(),
+        description="Path to SSL certificates",
+    )
+
+    is_dremio_cloud: bool = Field(
+        default=False,
+        description="Whether this is a Dremio Cloud instance",
+    )
+
+    dremio_cloud_region: Literal["US", "EMEA"] = Field(
+        default="US",
+        description="Dremio Cloud region ('US' or 'EMEA')",
+    )
+
+    @validator("authentication_method")
+    def validate_auth_method(cls, value):
+        allowed_methods = ["password", "PAT"]
+        if value not in allowed_methods:
+            raise ValueError(
+                f"authentication_method must be one of {allowed_methods}",
+            )
+        return value
+
+    @validator("password")
+    def validate_password(cls, value, values):
+        if values.get("authentication_method") == "PAT" and not value:
+            raise ValueError(
+                "Password (Personal Access Token) is required when using PAT authentication",
+            )
+        return value
+
+
+class ProfileConfig(GEProfilingConfig):
+    partition_profiling_enabled: bool = Field(
+        default=False,
+        description="Partition profiling disabled for Dremio.",
+    )
+    include_field_median_value: bool = Field(
+        default=False,
+        description="Median causes a number of issues in Dremio.",
+    )
+    query_timeout: int = Field(
+        default=300, description="Time before cancelling Dremio profiling query"
+    )
+
+    row_count: bool = True
+    column_count: bool = True
+    sample_values: bool = True
+
+
+class DremioSourceMapping(EnvConfigMixin, ConfigModel):
+    platform: Optional[str] = Field(
+        default=None,
+        description="Source connection made by Dremio (e.g. S3, Snowflake)",
+    )
+    platform_name: Optional[str] = Field(
```

Comment:
source_name ?

Reply:
This is the alias of the source in Dremio. Essentially we need this because you may have 2 S3 sources, one that is mapped to PROD URNs and another than is mapped to DEV URNs in datahub. It therefore allows you to differentiate the env, platform_instance, etc. with one Dremio but multiple upstream sources that may span platform_instances and environments.

Reply:
we could change this source_name as its more alighned which source and name we want to do upstream, as in config we can defined multiple source_mapping where we can specify the platform_instance

for eg:
```
source_mappings:
   - source: s3
      source_name: samples
      platform_instance: PROD
  - source: s3
      source_name: s3_test
      platform_instance: DEV
```

Reply:
Yes more like
```
source_mappings:
   -  source_name: samples
      platform: s3
      env: PROD
```
And this is required only if source type from dremio is not mapped to that of datahub in code OR if source platform is ingested with custom env or platform_instance in Datahub, right ?